README.md

```diff
@@ -21,35 +21,175 @@ The testing tasks are based on [istio.io Doc Tasks](https://istio.io/v1.9/docs/t
+By default, each attempt is logged, which typically results in a series of transient failures being shown in the log. 
+You can prevent maistra-test-tool from logging each attempt by setting the `LOG_FAILED_RETRY_ATTEMPTS` environment variable to `false`.
+The verbose output is mostly useful while writing or debugging tests. In CI systems, using the less-verbose form might be preffered.
+
+To disable verbose logging, run the tests as follows:
+```console
+LOG_FAILED_RETRY_ATTEMPTS=false make test
+```
+
+### Running tests in a container
+
+You can also run the test suite in a container, using the image `quay.io/maistra/maistra-test-tool:2.4`. 
+In addition to the environment variables explained above, you must also set the following environment variables:
+- `OCP_API_URL` - The URL of the OpenShift API server.
+- `OCP_TOKEN` - The authentication token for OpenShift. Alternatively, you can set the username and password via the `OCP_CRED_USR` and `OCP_CRED_PSW` environment variables.
+- `OCP_CRED_USR` - The username to use when logging into OpenShift.
+- `OCP_CRED_PSW` - The login password.
+
+
+For example, to run all the tests against your local CodeReady Containers cluster, run the following command:
+```console
+podman run -it \
+  --add-host api.crc.testing:$(crc ip) \
+  -e OCP_API_URL=https://api.crc.testing:6443 \
+  -e OCP_TOKEN=<token> \ 
+  quay.io/maistra/maistra-test-tool:2.4
+```
+
+Use the `-e` option to set the environment variables that affect the execution of the test suite, as described in the previous sections. 
 
-- Optionally to run all the test cases customizing the SMCP namespace and the SMCP name: A user can update the expected values in the `tests/test.env`.
+## Test Results
 
-- By default, the `tests/test.env` file uses `export OCP_ARCH=x86`
-    - For Power environment testing, a user can update the `tests/test.env` file `export OCP_ARCH=p`
-    - For Z environment testing, a user can update the `tests/test.env` file `export OCP_ARCH=z`
+Each time you run `make test`, a new directory named `result-<timestamp>` containing the results of the test run is created under `tests/`. 
+Additionally, a symbolic link named `result-latest` is updated to point to the latest result directory every time you run `make test`.
 
-- To run all the test cases, run the following command:
-  ```
-  make test
-  ```
+The result directory might look as the following example:
 
-- To run a single test case, run `make test <name>`. For example:
-  ```
-  make test TestFaultInjection
-  ```
+```console
+tests/
+└── result-20230203040506/          The root result directory for a particular test run.
+    ├── v2.2/                       Contains the results against the v2.3 version of the ServiceMeshControlPlane.
+    │   ├── failures-must-gather/   Contains must-gather snapshots of cluster resources for each failed test.
```

Comment:
This can be huge, would it be better to have the must gather optional with default=false?

Reply:
I don't think this is ever huge in test clusters. It's just a few YAMLs and logs. I don't think it will ever exceed 1MB.

If we disable it by default, then it won't be useful anymore, because you need must-gather the most when you don't expect it. If you enable it later and re-run the test, you may not see the same failure.

Reply:
Ok, I though it's bigger.

Reply:
oh, no.. you were right.. I just had a failure on ipv6 and the must-gather was 400mb... I'll investigate this.

Reply:
ah, the istio part is small (10MB), but the standard part is huge, yes.. I didn't know we capture everything.